buildpack.md

```diff
@@ -435,6 +446,60 @@ For each buildpack in the group, the lifecycle
 
 After analysis, the lifecycle MUST proceed to the build phase.
 
+
+## Phase #3: Image Extension
+
+### Purpose
+
+The purpose of the image extension phase is to apply dynamically-generated or static build-time and runtime modules that mutate the pre-build base image.
+
+During the image extension phase, an image extension might:
+
+1. Execute a pre-defined `Dockerfile` to install files in specific directories that require privilaged access.
+1. Dyanmically generate a `Dockerfile` based on the phase inputs and apply it.
+1. Execute a pre-defined `Dockerfile` to install system packages provided as input to the phase.
+
+### Process
+
+**GIVEN:**
+- The final ordered group of builimage extensions determined during the detection phase,
+- A directory containing application source code,
+- A `<output>` directory used to store generated artifacts
+- A shell, if needed,
+
+For each image extension in the group in order, the lifecycle MUST execute `/bin/build`.
+
+1. **If** the exit status of `/bin/build` is non-zero, \
+   **Then** the lifecycle MUST fail the phase.
+
+2. **If** the exit status of `/bin/build` is zero,
+   1. **If** there are additional image extensions in the group, \
+      **Then** the lifecycle MUST proceed to the next image extension's `/bin/build`.
+
+   2. **If** there are no additional image extension in the group, \
+      **Then** the lifecycle MUST proceed to the build phase.
+
+For each `/bin/build` executable in each image extension, the lifecycle:
+
+- MUST provide path arguments to `/bin/build` as described in the [Image Extension Interface](image-extension.md) section.
+- MUST configure the build environment as described in the [Environment](#environment) section.
+- MUST provide all `<plan>` entries that were required by any buildpack in the group during the detection phase with names matching the names that the buildpack provided.
+
+Correspondingly, each `/bin/build` executable:
+
+- MAY read from the `<app>` directory.
+- MUST NOT write to the `<app>` directory.
+- MAY read the build environment as described in the [Environment](#environment) section.
+- MAY read the Buildpack Plan.
+- MAY log output from the build process to `stdout`.
+- MAY emit error, warning, or debug messages to `stderr`.
+- SHOULD write BOM (Bill-of-Materials) entries to `<output>/launch.toml` describing any contributions to the app image.
+- MAY write key-value pairs to `<output>/launch.toml` that are provided as build args to run.Dockerfile or Dockerfile
+- SHOULD write build BOM entries to `<output>/build.toml` describing any contributions to the build environment.
+- MAY modify or delete any existing `<output>` directories.
+- MAY create new `<output>` directories.
+- MAY name any new `<output>` directories without restrictions except those imposed by the filesystem and the ones noted below.
```

Comment:
I'm not quite sure what to make of these bullets. I think they could be removed (when talking about `<layers>` they make sense but we don't really care about the output directory for extensions, only the files contained there).